controllers/disruption_controller.go

```diff
@@ -697,11 +705,16 @@ func (r *DisruptionReconciler) handleChaosPodTermination(ctx context.Context, in
 }
 
 func (r *DisruptionReconciler) updateTargetInjectionStatus(instance *chaosv1beta1.Disruption, chaosPod corev1.Pod, status chaostypes.DisruptionTargetInjectionStatus, since metav1.Time) {
-	targetInjection := instance.Status.TargetInjections[chaosPod.Labels[chaostypes.TargetLabel]]
+	targetInjections := instance.Status.TargetInjections[chaosPod.Labels[chaostypes.TargetLabel]]
+
+	disruptionKindName := chaostypes.DisruptionKindName(chaosPod.Labels[chaostypes.DisruptionKindLabel])
+
+	targetInjection := targetInjections.GetInjectionWithDisruptionKind(disruptionKindName)
```

Comment:
The `targetInjection`could be `nil`. I have to handle this cases when the `targetInjection` variable is `nil`